app/dojo.py

```diff
@@ -0,0 +1,164 @@
+    def add_person(cls, surname, firstname, category, wants_accomodation=None):
+        type_of_person = category.strip().lower()
+        try:
+            if type_of_person == "staff":
+                new_staff = Staff(surname, firstname)
+                if new_staff:
+                    print(("Staff {ns.surname} {ns.firstname} has been"
+                           + " successfully added").format(ns=new_staff))
+                    Staff.add_to_staff_list(new_staff)
+                    Person.add_to_map(new_staff)
+                    cls.allocate_room(new_staff)
+
+            elif type_of_person == "fellow":
+                new_fellow = Fellow(surname, firstname, True) \
+                    if wants_accomodation in ['y', 'Y'] \
+                    else Fellow(surname, firstname)
+                if new_fellow:
+                    print(("Fellow {nf.surname} {nf.firstname} has been"
+                           + " successfully added").format(nf=new_fellow))
+                    Fellow.add_to_fellow_list(new_fellow)
+                    Person.add_to_map(new_fellow)
+                    cls.allocate_room(new_fellow, wants_accomodation)
+            else:
+                print("Invalid type of person")
+
+        except WrongFormatException as e:
+            print(e)
+
+    @classmethod
+    def allocate_room(cls, person, wants_accomodation=None):
+        if(not(wants_accomodation in ['y', 'Y'])):
+            print(Office.allocate_office(person))
+        else:
+            print("{}{}".format(Office.allocate_office(person),
+                                LivingSpace.allocate_livingspace(person)))
+
+    @classmethod
+    def print_room(cls, room_name):
+        print("\n{}".format(Room.print_room_members(room_name)))
+
+    @classmethod
+    def print_allocations(cls, filename):
+        allocations = Room.get_allocations()
+        if filename is None:
+            print(allocations)
+        else:
+            try:
+                new_file = File.create_file(filename)
+                File.write(new_file, allocations)
+            except FileExistsError as e:
+                print(e)
+
+    @classmethod
+    def print_unallocated(cls, filename):
+        unallocated = Person \
+            .get_unallocated(Staff.get_unallocated_staff(),
+                             Fellow.get_unallocated_fellows())
+        if filename is None:
+            print(unallocated)
+        else:
+            try:
+                new_file = File.create_file(filename)
+                File.write(new_file, unallocated)
+            except FileExistsError as e:
+                print(e)
+
+    @classmethod
+    def reallocate_person(cls, identifier, room_name):
+        person_id = None
+        try:
+            person_id = int(identifier)
+            if Person.exist(person_id):
+                person = Person.id_map[person_id]
+                if Room.exists(room_name):
+                    room = Room.get_room_list()[room_name]
```

Comment:
`Room.get_room_list()` returns a dictionary, `[room_name]` is the key of the value i want to extract but notwithstanding i have refactored it. I implemented list comprehension in its replacement, so as to eliminate using a dictionary containing self referencing objects as key-value pair and also to follow standard variable naming convention because `get_room_list()` should return a list and not a dictionary.

Thanks for the feedback and insight.